libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -710,6 +710,24 @@ bool AP_Arming::gps_checks(bool report)
 }
 #endif  // AP_GPS_ENABLED
 
+bool AP_Arming::groundspeed_checks(bool report)
+{
+#if AP_GPS_ENABLED
+    if (!check_enabled(ARMING_CHECK_GPS)) {
+        return true;
+    }
+    // check only the first GPS:
+    const float gps_groundspeed = AP::gps().ground_speed();
+    const float ahrs_groundspeed = AP::ahrs().groundspeed();
+    if (fabsf(gps_groundspeed - ahrs_groundspeed) > 1.0f) {
```

Comment:
Would a 2D vector comparison be better? This can pass if there the same magnitude in opposite directions.

```suggestion
    const Vector2f gps_groundspeed = AP::gps().velocity().xy();
    const Vector2f ahrs_groundspeed = AP::ahrs().groundspeed_vector();
    if ((gps_groundspeed - ahrs_groundspeed).length_squared() > 1.0) {
```

Reply:
Makes the reporting a little harder of course....